### ISSUE TITLE
Scheduled removal sometimes occurs a tick late

### ISSUE THREAD

**Issue by ZakShearman** [Minestom/Minestom#624]
I regularly get this error with my damage indicators:

```
java.lang.ArrayIndexOutOfBoundsException: Index 16 out of bounds for length 16
	at pink.zak.minestom.towerdefence.model.mob.living.DamageIndicator.velocityTick(DamageIndicator.java:54)
	at net.minestom.server.entity.Entity.tick(Entity.java:534)
	at net.minestom.server.thread.TickThread.tick(TickThread.java:64)
	at net.minestom.server.thread.TickThread.run(TickThread.java:42)
```

This is because I have a 16 length array for my velocity and I get it by the `getAliveTicks` value.

When I create the entity I call `this.scheduleRemove(15, TimeUnit.SERVER_TICK);` so the entity should never live for 16 ticks, but it does.

See the file here for the code: https://github.com/ZakShearman/mc-tower-defence/blob/master/src/main/java/pink/zak/minestom/towerdefence/model/mob/living/DamageIndicator.java

If you notice that `scheduleRemove` is after the `setInstance` call, I've already tried with `scheduleRemove` before

**Comment by ZakShearman** [comment]
Not sure of the performance impact of using scheduleRemove for hundreds of entities either. I could instead just have a check in my tick method and then remove it after that time, but this is still an issue that should be fixed.

**Comment by TheMode** [comment]
Should be fixed in 5b699e0375402fa6a09e3d5e808ffd356289c0da

As for performance, I can only suggest using [async-profiler](https://github.com/jvm-profiling-tools/async-profiler)
